### PR TITLE
Build test for scipy 1.0.0

### PR DESCRIPTION
**Pull request by ocefpaf**
~~Do not merge! Just testing...~~

Based on the latest 1.0.0 release now :tada:

### CODE REVIEW

**Comment by ocefpaf** [comment]
@isuruf what is your take on the openblas version? 0.19 failed with the beta release and 0.20 went a while further in the tests.

**Comment by rgommers** [comment]
> @isuruf what is your take on the openblas version? 0.19 failed with the beta release and 0.20 went a while further in the tests.

Scipy wheels are now built and seem to work well for a recent OpenBLAS build (https://github.com/xianyi/OpenBLAS/commit/5f998ef) on all of Linux, OSX and Windows. Done in https://github.com/MacPython/scipy-wheels/pull/3

**Comment by ocefpaf** [comment]
> Scipy wheels are now built and seem to work well for a recent OpenBLAS build (https://github.com/xianyi/OpenBLAS/commit/5f998ef) on all of Linux, OSX and Windows. Done in https://github.com/MacPython/scipy-wheels/pull/3

@isuruf is the our openblas expert, I am not sure if `conda-forge`'s version has https://github.com/xianyi/OpenBLAS/commit/5f998ef I guess will probably need to update openblas first.

**Comment by ocefpaf** [comment]
Linux seems fine but `OS X` is segfaulting :unamused:

**Comment by ocefpaf** [comment]
Still segfaulting on OS X :-(

**Comment by chohner** [comment]
Any news on this?

**Comment by ocefpaf** [comment]
> Any news on this?

Nope :-/

@rgommers do you have any idea on what may be the cause for the segfault on `OS X`?

We can merge this to cash Linux and fix OS X later if the issue is too complicated.

**Comment by rgommers** [comment]
No idea, the build/test log is completely uninformative. If you push a commit with the change in ``run_test.py`` to have ``scipy.test(verbose=2)``, then at least we see where it crashes.

**Comment by ocefpaf** [comment]
> No idea, the build/test log is completely uninformative. If you push a commit with the change in ``run_test.py`` to have ``scipy.test(verbose=2)``, then at least we see where it crashes.

Thanks. Just did that. Let's see that happens.

**Comment by rgommers** [comment]
Crashes at:


    ::TestSplu::test_threads_parallel <- ../_t_env/lib/python3.6/site-packages/scipy/sparse/linalg/dsolve/tests/test_linsolve.py
    /Users/travis/miniconda3/conda-bld/scipy_1510595714136/test_tmp/conda_test_runner.sh: line 2: 
    11327 Segmentation fault: 11  "/Users/travis/miniconda3/conda-bld/scipy_1510595714136/_t_env/bin/python" 
    -s "/Users/travis/miniconda3/conda-bld/scipy_1510595714136/test_tmp/run_test.py"

So likely OpenBLAS related. IIRC there were some fixes from @pv in `test_threads_parallel`, however I can't locate them right now. Pauli, any suggestions?

**Comment by pv** [comment]
I don't think there are any relevant changes to that part of the code in v0.19.1 and v1.0.0.

**Comment by isuruf** [comment]
I can reproduce the same failure on CircleCI when `OPENBLAS_NUM_THREADS` is increased from 1 to 2.

**Comment by jakirkham** [comment]
Should already be set to `1`.

https://github.com/conda-forge/scipy-feedstock/blob/7693641f2d439f3f77fee2763188102f415521a4/recipe/run_test.py#L6

**Comment by isuruf** [comment]
Yes, what I'm saying is, when the value is 1, travis-ci fails and when the value is 2, both circle and travis-ci fails.

**Comment by isuruf** [comment]
OpenBLAS issue is fixed in https://github.com/conda-forge/openblas-feedstock/pull/38. I've restarted the tests on travis-ci

File: recipe/meta.yaml [hunk 1]
```diff
@@ -25,20 +25,20 @@ requirements:
...
-    - openblas 0.2.19|0.2.19*
+    - openblas 0.2.20|0.2.20*
     - numpy 1.9.*  # [not (win and py36)]
     - numpy 1.11.*  # [win and py36]
   run:
     - python
     - blas 1.1 {{ variant }}
     - libgfortran
-    - openblas 0.2.19|0.2.19*
+    - openblas 0.2.20|0.2.20*
```
Comment (isuruf) [inline recipe/meta.yaml]: Can you change this to `openblas >=0.2.20`?

Comment (ocefpaf) [reply recipe/meta.yaml]: Only run?

**Comment by ocefpaf** [comment]
Passing on my instance: https://travis-ci.org/ocefpaf/scipy-feedstock/builds/301863242

Thanks @isuruf! :tada:

File: recipe/meta.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/meta.yaml]: Yes. the run requirement can be relaxed that way.

**Comment by naoyak** [comment]
I think the numpy and scikit-learn recipes require corresponding updates to avoid an openblas conflict.

**Comment by isuruf** [comment]
@naoyak, can you send PRs for those? You can cc me and @ocefpaf

**Pull request merged**